src/Template/Provider/BaseProvider.php

```diff
@@ -87,15 +87,40 @@ public function listTemplateSources(): array
             self::PACKAGE_TYPE,
         );
 
-        foreach ($searchResult as ['name' => $packageName]) {
-            $package = $repository->findPackage($packageName, $constraint);
+        foreach ($searchResult as $result) {
+            $package = $repository->findPackage($result['name'], $constraint);
 
             if (null !== $package && self::PACKAGE_TYPE === $package->getType()) {
                 $templateSources[] = $this->createTemplateSource($package);
             }
         }
 
-        return $templateSources;
+        return $this->sortAbandonedPackagesToBottomOfTemplateSources($templateSources);
+    }
+
+    /**
+     * @param list<Template\TemplateSource> $templateSources
+     *
+     * @return list<Template\TemplateSource>
+     */
+    private function sortAbandonedPackagesToBottomOfTemplateSources(array $templateSources): array
+    {
+        $orderedTemplateSources = [];
+        $abandonedPackages = [];
+
+        foreach ($templateSources as $templateSource) {
+            if (
+                $templateSource->getPackage() instanceof Package\CompletePackageInterface
+                && $templateSource->getPackage()->isAbandoned()
+            ) {
+                $abandonedPackages[] = $templateSource;
+                continue;
+            }
+
+            $orderedTemplateSources[] = $templateSource;
+        }
+
+        return array_merge($orderedTemplateSources, $abandonedPackages);
```

Comment:
What about doing the whole merging right after the `foreach` in `listTemplateSources()` instead of creating a new method for this? This would avoid iterating twice over the found packages. Within `listTemplateSources()`, we could create two variables `$maintainedPackages` and `$abandonedPackages` and do the array merge right after the `foreach`, just as you did here in `sortAbandonedPackagesToBottomOfTemplateSources()`.